### PR TITLE
Add lua-vermelha as a optional Lua runtime

### PR DESCRIPTION
**Pull request by fsfod**
Mostly just builds the binary and copies it to the builds folder.

### CODE REVIEW

File: build_luav.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+#Build the various intresting flavors of LuaJIT
```
Comment (ltratt) [inline build_luav.sh]: Typo "interesting".

**Comment by ltratt** [comment]
I suggest we add this to `build_luajit.sh` and rename that (in this or another PR) to `build.sh`.

**Comment by fsfod** [comment]
I was just making it optional as a separate file

**Comment by ltratt** [comment]
I suggest that we make it non-optional as part of the build (it is quick to build, after all). Otherwise we'll end up with loads of `build_` scripts that we'll get confused by.

**Comment by fsfod** [comment]
Well travis runs that as well and it can't even build it even though it wasted over 10 mins trying

**Comment by ltratt** [comment]
I'm fine with guarding the build of this VM on Travis. You can check an environment variable for that IIRC.

**Comment by fsfod** [comment]
Well i should add it crashes the compiler on my Linux machine and on my laptop it fails to build with make -j only non parallel build works.

**Comment by ltratt** [comment]
Where are we with this?

**Comment by fsfod** [comment]
Was there a better branch we should be using i think you said before?

**Comment by ltratt** [comment]
If we guard the build, I don't think it matters?

**Comment by fsfod** [comment]
It was guarded for travis atleast. idk what option you want it under maybe BUILDALL

**Comment by ltratt** [comment]
Something like that sounds plausible.

**Comment by fsfod** [comment]
@ltratt I added a guard for it and and theres now main ./build.sh file setup in a previous commit

**Comment by ltratt** [comment]
Thanks. What does "theres now main ./build.sh file setup in a previous commit" mean though?

**Comment by fsfod** [comment]
i removed build_luav.sh and just added its contents into a new ./build.sh file that also builds luajit

**Comment by ltratt** [comment]
Ah, I see. I think what I was suggesting was renaming `build_luajit.sh` to `build.sh` and then putting the luav stuff in that new `build.sh`. Can we do something along those lines please?

**Comment by fsfod** [comment]
i wanted to keep build_luajit.sh seperate

**Comment by ltratt** [comment]
If we do that, we're going to end up with loads of `build_` files which are going to be confusing. It's not like a single `build.sh` is going to be huge.

**Comment by fsfod** [comment]
There won't be anymore any new ones will go in to build.sh

**Comment by ltratt** [comment]
OK, then merging them is even easier. [I'm always open to reconsidering my position, but "I want" isn't an adequate reason.]

**Comment by fsfod** [comment]
It will just be the main file i use for development i just want just something simple to run that i don't have remember to use any special command line.

**Comment by ltratt** [comment]
What special command line is needed for `build.sh`?

**Comment by fsfod** [comment]
none yet because you said to guard luav

**Comment by ltratt** [comment]
Perfect. Then there's nothing stopping merging the two files. If you do that, we should be able to merge this soon. Thanks!

**Comment by fsfod** [comment]
well i guess build_luajit was designed to optionally take a path to custom luajit dir

**Comment by ltratt** [comment]
It can still do that when it's merged in.

**Comment by fsfod** [comment]
@ltratt well there combined now

**Comment by ltratt** [comment]
Thanks!

One style comment: the indentation of `build.sh` is all over the place in parts e.g.:

```sh
+else
+    echo "Defaulting to ./luajit_repo/src as LuaJIT source path"
+    ljsrc=luajit_repo/src
+
+    if [ ! -d "luajit_repo" ]; then 
+      git clone https://github.com/softdevteam/LuaJIT luajit_repo
+    fi
```

Can you add a separate commit which fixes this?

**Comment by fsfod** [comment]
@ltratt fixed

**Comment by ltratt** [comment]
Thanks. If you can squash this down (please leave the indentation commit separate), then I can merge.

**Comment by fsfod** [comment]
@ltratt why do you want to keep the indentation commit the file always has lots of churn before that

**Comment by ltratt** [comment]
To reduce the churn.

**Comment by fsfod** [comment]
@ltratt well the code is squashed

**Pull request merged**